test/XRP/default-xrp-client-test.ts

```diff
@@ -604,4 +607,88 @@ describe('Default Xpring Client', function (): void {
+
+    // THEN the returned transaction is as expected.
+    assert.deepEqual(
+      transaction,
+      XRPTransaction.from(testGetTransactionResponseProto),
+    )
+  })
+
+  it('Get Payment - failing network request with NOT_FOUND error', function (done) {
+    // GIVEN a DefaultXRPClient with mocked networking that will fail to retrieve a transaction w/ NOT_FOUND error code.
+    const notFoundError = new FakeGRPCError(
+      'FakeGRPCError: account not found',
+      grpcStatusCode.NOT_FOUND,
+    )
+    const fakeNetworkResponses = new FakeXRPNetworkClientResponses(
+      FakeXRPNetworkClientResponses.defaultAccountInfoResponse(),
+      FakeXRPNetworkClientResponses.defaultFeeResponse(),
+      FakeXRPNetworkClientResponses.defaultSubmitTransactionResponse(),
+      notFoundError,
+    )
+    const fakeNetworkClient = new FakeXRPNetworkClient(fakeNetworkResponses)
+    const xrpClient = new DefaultXRPClient(fakeNetworkClient)
+
+    // WHEN a transaction is requested, THEN the error is re-thrown.
+    xrpClient.getPayment(transactionHash).catch((error) => {
+      assert.typeOf(error, 'Error')
+      assert.equal(error.code, grpcStatusCode.NOT_FOUND)
+      done()
+    })
+  })
+
+  it('Get Payment - malformed payment transaction', async function (): Promise<
+    void
+  > {
+    // GIVEN a DefaultXRPClient with mocked networking that will return a malformed payment transaction.
+    const fakeNetworkResponses = new FakeXRPNetworkClientResponses(
+      FakeXRPNetworkClientResponses.defaultAccountInfoResponse(),
+      FakeXRPNetworkClientResponses.defaultFeeResponse(),
+      FakeXRPNetworkClientResponses.defaultSubmitTransactionResponse(),
+      testInvalidGetTransactionResponseProto,
+    )
+    const fakeNetworkClient = new FakeXRPNetworkClient(fakeNetworkResponses)
+    const xrpClient = new DefaultXRPClient(fakeNetworkClient)
+
+    // WHEN a transaction is requested.
+    const transaction = await xrpClient.getPayment(transactionHash)
+
+    // THEN the result is undefined
+    assert.isUndefined(transaction)
+  })
+
+  it('Get Payment - unsupported transaction type', async function (): Promise<
+    void
+  > {
+    // GIVEN a DefaultXRPClient with mocked networking that will return an unsupported transaction type.
+    const fakeNetworkResponses = new FakeXRPNetworkClientResponses(
+      FakeXRPNetworkClientResponses.defaultAccountInfoResponse(),
+      FakeXRPNetworkClientResponses.defaultFeeResponse(),
+      FakeXRPNetworkClientResponses.defaultSubmitTransactionResponse(),
+      testInvalidGetTransactionResponseProtoUnsupportedType,
+    )
+    const fakeNetworkClient = new FakeXRPNetworkClient(fakeNetworkResponses)
+    const xrpClient = new DefaultXRPClient(fakeNetworkClient)
+
+    // WHEN a transaction is requested.
+    const transaction = await xrpClient.getPayment(transactionHash)
+
+    // THEN the result is undefined
+    assert.isUndefined(transaction)
+  })
```

Comment:
Is it worth adding an integration test that isn't mocked? I'm not sure how you guys make the decision for what to mock vs not mock?

Reply:
So everything in this file is mocked and I believe considered a unit test?  The integration tests for `DefaultXRPClient` are essentially the integration tests for `XRPClient` because the `XRPClient` decorates an underlying `DefaultXRPClient`.